Signal/src/ViewControllers/HomeView/HomeViewController.m

```diff
@@ -859,11 +875,44 @@ - (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)
     return YES;
 }
 
-#pragma mark - SearchResultsUpdating
+#pragma mark - UISearchBarDelegate
+
+- (void)searchBarTextDidBeginEditing:(UISearchBar *)searchBar
+{
+    [self.tableView setContentOffset:CGPointZero animated:NO];
+
+    [self updateSearchResultsVisibility];
+}
+
+- (void)searchBarTextDidEndEditing:(UISearchBar *)searchBar
+{
+    [self updateSearchResultsVisibility];
+}
+
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText
+{
+    [self updateSearchResultsVisibility];
+}
+
+- (void)searchBarSearchButtonClicked:(UISearchBar *)searchBar
+{
+    [self updateSearchResultsVisibility];
+}
+
+- (void)searchBarCancelButtonClicked:(UISearchBar *)searchBar
+{
+    self.searchBar.text = nil;
+
+    [self updateSearchResultsVisibility];
+}
 
-- (void)updateSearchResultsForSearchController:(UISearchController *)searchController
+- (void)updateSearchResultsVisibility
 {
-    [self.searchResultsController updateSearchResultsWithSearchText:self.searchController.searchBar.text];
+    OWSAssertIsOnMainThread();
+
+    [self.searchResultsController updateSearchResultsWithSearchText:self.searchBar.text];
+    BOOL isSearching = self.searchBar.text.length > 0;
```

Comment:
Should we use ows_stripped,  since it's less obvious when you're in "search" mode now, it might be good to only trigger the search results view when there is visible content in the search bar.

Reply:
done.